newlib/libc/sys/vita/syscalls.c

```diff
@@ -382,3 +382,40 @@ _stat_r(struct _reent *reent, const char *path, struct stat *st)
 	reent->_errno = 0;
 	return 0;
 }
+
+int
+_ftruncate_r(struct _reent *reent, int fd, off_t length)
+{
+	struct SceIoStat stat = {st_size: length};
```

Comment:
- This syntax is [obsolete since GCC 2.5](https://gcc.gnu.org/onlinedocs/gcc/Designated-Inits.html).
- Designated initializers are not supported in C++. There is a proposal though: http://open-std.org/JTC1/SC22/WG21/docs/papers/2016/p0329r0.pdf

Reply:
I always hesitated between the two declarations versions.
I won't anymore, thanks for the info.